field/TinyLang/Field/Rename.hs

```diff
@@ -28,11 +34,11 @@ renameVarM var = do
         Nothing   -> var
         Just uniq -> setUnique uniq var
 
-withRenamedStatementM :: Statement f -> (Statement f -> RenameM c) -> RenameM c
-withRenamedStatementM (ELet (UniVar uni var) def) cont = do
+renameStatementM :: Statement f -> RenameM (Statement f)
+renameStatementM (ELet (UniVar uni var) def) = do
     defRen <- renameExprM def
-    withFreshenedVar var $ \varFr -> cont $ ELet (UniVar uni varFr) defRen
-withRenamedStatementM (EAssert expr)              cont = renameExprM expr >>= cont . EAssert
+    withFreshenedVar var $ \varFr -> pure $ ELet (UniVar uni varFr) defRen 
```

Comment:
Could you also add a property test where we generate a program, rename it and evaluate and then check if the result matches with just evaluating the program, without the renaming. Such a test has to fail here. `withFreshenedVar` updates the local context of the continuation, and here you just immediately leave that context with `pure`, so all the statements get renamed in the old context, not in the new one.

Reply:
Done, still need to add tests.

Reply:
Is there an easy way to generate an environment to close the program?

Reply:
See [these lines](https://github.com/effectfully/tiny-lang/blob/master/field/TinyLang/Field/Generator.hs#L444-L457).